### PR TITLE
Add garfield accounting backend

### PR DESCRIPTION
**Pull request by TehMillhouse**
TODO: ensure we don't use an out-of-date tax rate when running for prolonged time spans

### CODE REVIEW

**Comment by TehMillhouse** [comment]
(I fixed the cashing thing)

File: accounting.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    cash_box_ids = {cshbx.cash_box_name: cshbx.cash_box_id for cshbx in cash_boxes}
+
+    # Ermäßigte Umsatzsteuer: tax_group id 2
+    _cached_exam_sale_tax_id = None
+    _tax_id_end_date = None
+    def _exam_sale_tax_id():
+        global _tax_id_end_date, _cached_exam_sale_tax_id
+        # Check if our cached tax has expired, if so, fetch it
+        if not _tax_id_end_date or _tax_id_end_date < datetime.date.today():
+            tax = db.session.query(Base.classes.tax_mappings)\
```
Comment (Kha) [inline accounting.py]: `garfield.tax_find`?

File: accounting.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+                cash_box_ids[cashbox],
+                sqlalchemy.sql.func.current_date())
+        return db.session.execute(proc).scalar()
+
+    def _user_id(user_name: str):
+        return db.session.query(Base.classes.users.user_id)\
+                .filter(Base.classes.users.user_name == user_name).scalar()
+
+    def log_donation(amount: int, cashbox: str):
+        proc = procs.donation_accept(cash_box_ids[cashbox], 'MONEY', amount / 100)
```
Comment (Kha) [inline accounting.py]: I like the part that says `'MONEYZ'`

File: accounting.py [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+        db.session.execute(proc)
+
+    def _cash_box_log_entry(user, cashbox: str, amount: int, action: str):
+        return Base.classes.cash_box_log(
+                cash_box_log_timestamp=sqlalchemy.sql.func.now(),
```
Comment (Kha) [inline accounting.py]: default value of the DB column

**Comment by TehMillhouse** [comment]
RFC on the newest state. The whole `__table__` thing is fairly stinky, but it avoids both literal SQL statement format strings and python-side construction of mapped instances.

**Pull request merged**